devops/continuous_integration/packages_pipeline/multi_configuration.rst

```diff
@@ -0,0 +1,178 @@
+
+
+.. code-block:: bash
+    :caption: Debug build
+
+    $ conan create . --build="missing:ai/*" -s build_type=Debug --format=json > graph.json
+    $ conan list --graph=graph.json --graph-binaries=build --format=json > upload_debug.json
+    $ conan remote add packages "<url-packages-repo>" -f  # Can be ommitted, it was defined above
+    $ conan upload -l=upload_debug.json -r=packages -c --format=json > upload_debug.json
+
+
+When both Release and Debug configuration finish successfully, we would have these packages in the repositories:
+
+.. graphviz::
+    :align: center
+
+    digraph repositories {
+        node [fillcolor="lightskyblue", style=filled, shape=box]
+        rankdir="LR"; 
+        subgraph cluster_0 {
+                label="Packages server";
+                style=filled;
+                color=lightgrey;
+                subgraph cluster_1 {
+                label = "packages\n repository" 
+                shape = "box";
+                style=filled;
+                color=lightblue;
+                "packages" [style=invis];
+                "ai/1.1.0\n (Release)";
+                "ai/1.1.0\n (Debug)";
+                }
+                subgraph cluster_2 {
+                label = "products\n repository" 
+                shape = "box";
+                style=filled;
+                color=lightblue;
+                "products" [style=invis];
+                } 
+                subgraph cluster_3 {
+                rankdir="BT";
+                shape = "box";
+                label = "develop repository";
+                color=lightblue;
+                rankdir="BT";
+        
+                node [fillcolor="lightskyblue", style=filled, shape=box]
+                "game/1.0" -> "engine/1.0" -> "ai/1.0" -> "mathlib/1.0";
+                "engine/1.0" -> "graphics/1.0" -> "mathlib/1.0";
+                "mapviewer/1.0" -> "graphics/1.0";
+                "game/1.0" [fillcolor="lightgreen"];
+                "mapviewer/1.0" [fillcolor="lightgreen"];
+                }
+                {
+                edge[style=invis];
+                "packages" -> "products" -> "game/1.0" ; 
+                rankdir="BT";    
+                }
+        }
+    }
+
+
+If the build of all configurations for ``ai/1.1.0`` were succesfull, then the ``packages pipeline`` can proceed and promote
+them to the ``products`` repository:
+
+.. code-block:: bash
+    :caption: Promoting from packages->product
+
+    # aggregate the package list
+    $ conan pkglist merge -l upload_release.json -l upload_debug.json --format=json > promote.json
+
+    $ conan remote add packages "<url-packages-repo>" -f  # Can be ommitted, it was defined above
+    $ conan remote add products "<url-products-repo>" -f  # Can be ommitted, it was defined above
+
+    # Promotion with Artifactory CE (slow, can be improved with art:promote)
+    $ conan download --list=promote.json -r=packages --format=json > promote.json
```

Comment:
```suggestion
    $ conan download --list=merged.json -r=packages --format=json > promote.json
```
same problem with the file being deleted before read

Reply:
Oh, maybe it works differently with our ``run(..., redirect_output=filename)`` implementation. 
Understood, I need to change this. I will change it first in the ``examples2`` repo, then update it here.